File: solutionFiles/MergedTwoList.py
import sys
import json

def merge_two_sorted_lists(l1, l2):
    merged_list = []
    i, j = 0, 0

    # Traverse both lists and append the smaller element to the merged list
    while i < len(l1) and j < len(l2):
        if l1[i] <= l2[j]:
            merged_list.append(l1[i])
            i += 1
        else:
            merged_list.append(l2[j])
            j += 1

    # Append remaining elements, if any
    while i < len(l1):
        merged_list.append(l1[i])
        i += 1

    while j < len(l2):
        merged_list.append(l2[j])
        j += 1

    return merged_list

def main():
    # Assuming the input is passed as the first command line argument
    input_json = sys.argv[1]

    # Parse the JSON input
    data = json.loads(input_json)
    l1 = data.get('list1', [])
    l2 = data.get('list2', [])

    # Merge the two sorted lists
    merged_list = merge_two_sorted_lists(l1, l2)

    # Prepare the result in the desired format
    result = {"result": merged_list}

    # Print the result as a JSON string
    output_json = json.dumps(result)
    print(output_json)

if __name__ == "__main__":
    main()

File: solutionFiles/MergedTwoList.js
function mergeTwoSortedLists(list1, list2) {
    let mergedList = [];
    let i = 0, j = 0;

    // Traverse both lists and append the smaller element to the merged list
    while (i < list1.length && j < list2.length) {
        if (list1[i] <= list2[j]) {
            mergedList.push(list1[i]);
            i++;
        } else {
            mergedList.push(list2[j]);
            j++;
        }
    }

    // Append remaining elements, if any
    while (i < list1.length) {
        mergedList.push(list1[i]);
        i++;
    }

    while (j < list2.length) {
        mergedList.push(list2[j]);
        j++;
    }

    return mergedList;
}

function main() {
    // Assuming the input is passed as the first command line argument
    const inputJson = process.argv[2];

    // Parse the JSON input
    const data = JSON.parse(inputJson);
    const list1 = data.list1 || [];
    const list2 = data.list2 || [];

    // Merge the two sorted lists
    const mergedList = mergeTwoSortedLists(list1, list2);

    // Prepare the result in the desired format
    const result = { result: mergedList };

    // Print the result as a JSON string
    console.log(JSON.stringify(result));
}

main();
